Sources/MongoSwift/APM.swift

```diff
@@ -1,24 +1,32 @@
 import Foundation
 import libmongoc
 
-/// A struct representing a server connection, consisting of a host and port.
-public struct ConnectionId {
-    let host: String
-    let port: UInt16
-
-    internal init(_ hostList: UnsafePointer<mongoc_host_list_t>) {
-        var hostData = hostList.pointee
-        self.host = withUnsafeBytes(of: &hostData.host) { (rawPtr) -> String in
-            let ptr = rawPtr.baseAddress!.assumingMemoryBound(to: CChar.self)
-            return String(cString: ptr)
-        }
-        self.port = hostData.port
-    }
+/// A protocol for monitoring events to implement, specifying their type and name.
+public protocol MongoEvent {
```

Comment:
I changed this to public so that it is actually useful to users. for example, they might want to store events in an array of `MongoEvent`. 
the static `eventName` property will also allow them to do something like
```swift
let observer = center.addObserver(forName: CommandStartedEvent.eventName, object: nil, queue: nil) { (notif) in
    // do stuff
}
```

and then added a separate protocol for the internal initializer so we can still post everything through `postNotification`, which requires that initializer exists

Reply:
very nice!

Reply:
very nice!

Reply:
very nice!